### ISSUE TITLE
Docker image file generated by parameterized_docker_build.sh fails (3)

### ISSUE THREAD

**Issue by SaintNazaire** [tensorflow/tensorflow#13784]
**Issue**

Fixed previous docker install issue [#13379](https://github.com/tensorflow/tensorflow/issues/13379)

Executing parameterized_docker_build.sh generates new docker file but image fails to turn into an docker image, throws error message.

**System information**

- I have used a stock example script provided in TensorFlow
- Windows 10 professional
- TensorFlow install as docker image tensorflow/tensorflow:1.3.0-devel-py3
- TensorFlow version 1.3
- Python version 3
- Bazel version 0.5.0
- CUDA/cuDNN version not relevant (CPU install)
- GPU model and memory not relevant (CPU install)
- Docker CE 17.09.0-ce

`$which docker`
> /usr/bin/docker

**Command triggering issue**

`$/tensorflow/tensorflow/tools/docker/parameterized_docker_build.sh`

**Error messages**

> Required build parameters:
>   TF_DOCKER_BUILD_TYPE=cpu
>   TF_DOCKER_BUILD_IS_DEVEL=no
>   TF_DOCKER_BUILD_DEVEL_BRANCH=
> 
> Optional build parameters:
>   TF_DOCKER_BUILD_CENTRAL_PIP=http://ci.tensorflow.org/view/Nightly/job/nightly-python35-linux-cpu/lastSuccessfulBuild/artifact/pip_test/whl/tensorflow-1.head-cp35-cp35m-manylinux1_x86_64.whl
>   TF_DOCKER_BUILD_IMAGE_NAME=
>   TF_DOCKER_BUILD_VERSION=
>   TF_DOCKER_BUILD_PORT=
>   TF_DOCKER_BUILD_PUSH_CMD=
> 
> FINAL_IMAGE_NAME: tensorflow/tensorflow
> FINAL_TAG: latest
> Original Dockerfile: /tensorflow/tensorflow/tools/docker/Dockerfile
> 
> 
> Docker build will occur in temporary directory: /tmp/tmp.nBIOwyaShf
> Downloading pip wheel from: http://ci.tensorflow.org/view/Nightly/job/nightly-python35-linux-cpu/lastSuccessfulBuild/artifact/pip_test/whl/tensorflow-1.head-cp35-cp35m-manylinux1_x86_64.whl
> 
> Modified Dockerfile at: /tmp/tmp.nBIOwyaShf/Dockerfile
> 
> Building docker image with image name and tag: /tensorflow:latest
> invalid argument "/tensorflow:latest" for t: invalid reference format
> See 'docker build --help'.
> FAIL: docker build of /tensorflow:latest with Dockerfile /tmp/tmp.nBIOwyaShf/Dockerfile failed

**Inspecting the resulting docker file for further introspection**

`$cat /tmp/tmp.nBIOwyaShf/Dockerfile`

> FROM ubuntu:16.04
> 
> MAINTAINER Craig Citro <craigcitro@google.com>
> 
> RUN apt-get update && apt-get install -y --no-install-recommends \
>         build-essential \
>         curl \
>         libfreetype6-dev \
>         libpng12-dev \
>         libzmq3-dev \
>         pkg-config \
>         python \
>         python-dev \
>         rsync \
>         software-properties-common \
>         unzip \
>         && \
>     apt-get clean && \
>     rm -rf /var/lib/apt/lists/*
> 
> RUN curl -O https://bootstrap.pypa.io/get-pip.py && \
>     python get-pip.py && \
>     rm get-pip.py
> 
> RUN pip --no-cache-dir install \
>         Pillow \
>         h5py \
>         ipykernel \
>         jupyter \
>         matplotlib \
>         numpy \
>         pandas \
>         scipy \
>         sklearn \
>         && \
>     python -m ipykernel.kernelspec
> 
> RUN pip --no-cache-dir install http://ci.tensorflow.org/view/Nightly/job/nightly-python35-linux-cpu/lastSuccessfulBuild/artifact/pip_test/whl/tensorflow-1.head-cp35-cp35m-manylinux1_x86_64.whl
> 
> 
> COPY jupyter_notebook_config.py /root/.jupyter/
> 
> COPY notebooks /notebooks
> 
> COPY run_jupyter.sh /
> 
> EXPOSE 6006
> EXPOSE 8888
> 
> WORKDIR "/notebooks"
> 
> CMD ["/run_jupyter.sh", "--allow-root"]

**Additional introspection with docker image file**

Copied and pasted the parameterized_docker_build.sh output docker file content and tried to execute on docker hub which fails.

Docker hub log reports yields the following additional error information:

> Step 6/9 : RUN pip --no-cache-dir install http://ci.tensorflow.org/view/Nightly/job/nightly-python35-linux-cpu/lastSuccessfulBuild/artifact/pip_test/whl/tensorflow-1.head-cp35-cp35m-manylinux1_x86_64.whl
> 
>  ---> Running in fd0d49a714e4
> 
> [91mtensorflow-1.head-cp35-cp35m-manylinux1_x86_64.whl is not a supported wheel on this platform.
> [0m
> Removing intermediate container fd0d49a714e4
> 
> The command '/bin/sh -c pip --no-cache-dir install http://ci.tensorflow.org/view/Nightly/job/nightly-python35-linux-cpu/lastSuccessfulBuild/artifact/pip_test/whl/tensorflow-1.head-cp35-cp35m-manylinux1_x86_64.whl' returned a non-zero code: 1

**Comment by andydavis1** [comment]
@caisq Can you take a look?

**Comment by SaintNazaire** [comment]
**Complementary test for docker image file introspection through docker hub**

Visited Tensorflow's [Jenkins automation server](http://ci.tensorflow.org/view/Nightly/job/nightly-python35-linux-cpu/lastSuccessfulBuild/artifact/pip_test/whl/ ), found out that there was another version named tensorflow-1.head-cp35-cp35m-linux_x86_64.whl instead of tensorflow-1.head-cp35-cp35m-manylinux1_x86_64.whl

Edited the docker file to substitute one by the other which triggered the exact same error when trying to build docker image via docker hub.

Docker hub log issued following error message:

> Step 6/9 : RUN pip --no-cache-dir install http://ci.tensorflow.org/view/Nightly/job/nightly-python35-linux-cpu/lastSuccessfulBuild/artifact/pip_test/whl/tensorflow-1.head-cp35-cp35m-linux_x86_64.whl
> 
>  ---> Running in ec97557752ad
> 
> [91mtensorflow-1.head-cp35-cp35m-linux_x86_64.whl is not a supported wheel on this platform.
> [0m
> Removing intermediate container ec97557752ad
> 
> The command '/bin/sh -c pip --no-cache-dir install http://ci.tensorflow.org/view/Nightly/job/nightly-python35-linux-cpu/lastSuccessfulBuild/artifact/pip_test/whl/tensorflow-1.head-cp35-cp35m-linux_x86_64.whl' returned a non-zero code: 1

**Comment by SaintNazaire** [comment]
Seems the script installs Python2 then install a Python3 version of Tensorflow which is obviously wrong.
Problem seems to be with TF_DOCKER_BUILD_PYTHON_VERSION=PYTHON3 which doesn't trigger properly the selection of a Python3 version of Tensorflow via a pip3 install.

**Comment by rodhammond** [comment]
Sorry if I've misunderstood your issue, but think this will solve your problems. I'm using Ubuntu 14.04 and the tensorflow/tensorflow docker image. The NV_GPU option allows GPU access of my two 980 cards to the docker container. I think the overall problem is caused by an unset ENV $USER variable.

1. Launch the container using directory mapping to avoid docker-in-docker ulimit and process pid/socket issues:

NV_GPU=0,1 nvidia-docker run --rm -it --ulimit nproc=1024:2048 -v /var/lib/docker:/var/lib/docker/ -v /var/run/docker.sock:/var/run/docker.sock -v /var/run/docker.pid:/var/run/docker-ssd.pid -v /var/run/docker.pid:/var/run/docker.pid --name jupyter_tensorflow -p 8888:8888 gcr.io/tensorflow/tensorflow:latest-devel-gpu /bin/bash

2. Once at the bash prompt, install the following:

apt-get update && \
    apt-get install \
    apt-transport-https \
    ca-certificates \
    curl \
    software-properties-common && \
    curl -fsSL https://download.docker.com/linux/ubuntu/gpg | apt-key add - && \
    add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu \
    $(lsb_release -cs) stable" && \
    apt-get update && \
    apt-get install docker-ce && \
    apt-get install iputils-ping && \
    apt-get install nano && \
    which docker

3. Set ENV variables:

rm -rf /tmp/tmp*
cd /tensorflow/tensorflow/tools/docker

export USER=tensorflow

export TF_DOCKER_BUILD_IS_DEVEL=NO
export TF_DOCKER_BUILD_TYPE=CPU
export TF_DOCKER_BUILD_PYTHON_VERSION=PYTHON2
export NIGHTLY_VERSION="1.head"
export TF_DOCKER_BUILD_IMAGE_NAME=tensorflow
export TF_DOCKER_BUILD_VERSION=latest

export TF_DOCKER_BUILD_CENTRAL_PIP=$(echo ${TF_DOCKER_BUILD_PYTHON_VERSION} | sed s^PYTHON2^http://ci.tensorflow.org/view/Nightly/job/nightly-matrix-cpu/TF_BUILD_IS_OPT=OPT,TF_BUILD_IS_PIP=PIP,TF_BUILD_PYTHON_VERSION=${TF_DOCKER_BUILD_PYTHON_VERSION},label=cpu-slave/lastSuccessfulBuild/artifact/pip_test/whl/tensorflow-${NIGHTLY_VERSION}-cp27-cp27mu-manylinux1_x86_64.whl^ | sed s^PYTHON3^http://ci.tensorflow.org/view/Nightly/job/nightly-python35-linux-cpu/lastSuccessfulBuild/artifact/pip_test/whl/tensorflow-${NIGHTLY_VERSION}-cp35-cp35m-manylinux1_x86_64.whl^)

4. Execute the parameterized_docker_build script to build a local image with the  above settings:

chmod +x /tensorflow/tensorflow/tools/docker/parameterized_docker_build.sh
/tensorflow/tensorflow/tools/docker/parameterized_docker_build.sh

Hope this helps.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by SaintNazaire** [comment]
Could not replicate issue with clean install, closing.
